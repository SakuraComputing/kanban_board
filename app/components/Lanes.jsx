import React, { Component } from 'react';

export default class MyComponent extends Component {
  render() {
    return (
      <div className="lanes">
        lanes should go here
      </div>
    );
  }
}



