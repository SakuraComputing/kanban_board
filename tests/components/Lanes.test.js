import React from 'react';
import { shallow } from 'enzyme';
import Lanes from '../../app/components/Lanes';


describe('Lanes Component', () => {
    let wrapper, lanes;

    beforeEach(() => {
        lanes = [jest.fn()];
        wrapper = shallow(<Lanes lanes={lanes}/>);
    });

    it('should exist', () => {
        expect(wrapper).toBeTruthy();
    });

    it('should render the form correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
