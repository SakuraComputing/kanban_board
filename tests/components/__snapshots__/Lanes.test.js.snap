// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lanes Component should render the form correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    lanes={
      Array [
        [MockFunction],
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <DropTarget(Lane)
          className="lane"
          lane={[MockFunction]}
        >
          <LaneHeader
            name="mockConstructor"
          />
          <LaneNotes
            notes={undefined}
          />
        </DropTarget(Lane)>,
      ],
      "className": "lanes",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <LaneHeader
              name="mockConstructor"
            />,
            <LaneNotes
              notes={undefined}
            />,
          ],
          "className": "lane",
          "lane": [MockFunction],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "mockConstructor",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "notes": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <DropTarget(Lane)
            className="lane"
            lane={[MockFunction]}
          >
            <LaneHeader
              name="mockConstructor"
            />
            <LaneNotes
              notes={undefined}
            />
          </DropTarget(Lane)>,
        ],
        "className": "lanes",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <LaneHeader
                name="mockConstructor"
              />,
              <LaneNotes
                notes={undefined}
              />,
            ],
            "className": "lane",
            "lane": [MockFunction],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "mockConstructor",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "notes": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
